src/components/plots/areaPlot.ts

```diff
@@ -89,6 +116,37 @@ export module Plots {
       return attrToProjector;
     }
 
+    public getAllSelections(datasets = this.datasets(), exclude = false) {
+      var allSelections = super.getAllSelections(datasets, exclude)[0];
+      if (exclude) {
+        datasets = this.datasets().filter((dataset) => datasets.indexOf(dataset) < 0);
+      }
+      var lineDrawers = datasets.map((dataset) => this._lineDrawers.get(dataset))
+                                .filter((drawer) => drawer != null);
+      lineDrawers.forEach((ld, i) => allSelections.push(ld._getSelection(i).node()));
+      return d3.selectAll(allSelections);
+    }
+
+    public getAllPlotData(datasets = this.datasets()): Plots.PlotData {
+      var allPlotData = super.getAllPlotData(datasets);
+      var allElements = allPlotData.selection[0];
+
+      this._keysForDatasets(datasets).forEach((datasetKey) => {
+        var plotDatasetKey = this._key2PlotDatasetKey.get(datasetKey);
+        if (plotDatasetKey == null) { return; }
+        var drawer = plotDatasetKey.drawer;
+        var dataset = plotDatasetKey.dataset;
+        plotDatasetKey.dataset.data().forEach((datum: any, index: number) => {
+          var pixelPoint = this._pixelPoint(datum, index, dataset);
+          if (pixelPoint.x !== pixelPoint.x || pixelPoint.y !== pixelPoint.y) {
+            return;
+          }
+          allElements.splice(index * 2 + 1, 0, drawer._getSelection(index).node());
```

Comment:
Why is it spliced to this location? If the `.line` `<path>`s are supposed to be at the end (to parallel `getAllSelections`), they should just be `push`ed right?